nucypher/blockchain/eth/agents.py

```diff
@@ -543,6 +544,139 @@ def bond_operator(self, staking_provider: ChecksumAddress, operator: ChecksumAdd
+            FAILED = 4
+            FINAL = 5
+
+        @dataclass
+        class Performance:
+            node: ChecksumAddress
+            aggregated: bool
+            transcript: bytes
+
+        id: int
+        status: Status
+        init_timestamp: int
+        total_transcripts: int
+        total_aggregations: int
+        performances: List = field(default_factory=list)
+
+        @property
+        def nodes(self):
+            return [p.node for p in self.performances]
+
+        @property
+        def transcripts(self) -> List[bytes]:
+            transcripts = list()
+            for p in self.performances:
+                if p.aggregated:
+                    raise RuntimeError(f"{p.node[:8]} transcript is already aggregated")
+                transcripts.append(p.transcript)
+            return transcripts
+
+        @property
+        def aggregated_transcripts(self) -> List[bytes]:
+            transcripts = list()
+            for p in self.performances:
+                if not p.aggregated:
+                    raise RuntimeError(f"{p.node[:8]} transcript not aggregated")
+                transcripts.append(p.transcript)
+            return transcripts
+
+        @property
+        def shares(self) -> int:
+            return len(self.nodes)
+
+    @contract_api(CONTRACT_CALL)
+    def get_ritual(self, ritual_id: int) -> Ritual:
+        result = self.contract.functions.rituals(int(ritual_id)).call()
+        ritual = self.Ritual(
+            id=ritual_id,
+            status=result[0],
+            init_timestamp=result[1],
+            total_transcripts=result[2],
+            total_aggregations=result[3],
+            performances=[],
+        )
+        return ritual
+
+    @contract_api(CONTRACT_CALL)
+    def get_performances(self, ritual_id: int) -> List[Ritual.Performance]:
+        result = self.contract.functions.getPerformances(ritual_id).call()
+        performances = list()
+        for r in result:
+            performance = self.Ritual.Performance(
+                node=ChecksumAddress(r[0]), aggregated=r[1], transcript=bytes(r[2])
+            )
+            performances.append(performance)
+        return performances
+
+    @contract_api(CONTRACT_CALL)
+    def number_of_rituals(self) -> int:
+        result = self.contract.functions.numberOfRituals().call()
+        return result
+
+    @contract_api(TRANSACTION)
+    def initiate_ritual(self, nodes: List[ChecksumAddress], transacting_power: TransactingPower) -> TxReceipt:
+        contract_function: ContractFunction = self.contract.functions.initiateRitual(nodes=nodes)
+        receipt = self.blockchain.send_transaction(contract_function=contract_function,
+                                                   transacting_power=transacting_power)
+        return receipt
+
+    @contract_api(TRANSACTION)
+    def post_transcript(self, ritual_id: int, transcript: bytes, node_index: int, transacting_power: TransactingPower) -> TxReceipt:
```

Comment:
I'm exposing a `Transcript` class from `ferveo-python`. I think it would be useful to provide a serialization method compatible with whatever this contract call is expecting. I already have `to_bytes` implementation but not sure it's going to cut it.